### PR TITLE
Added hasMessageMatching assertion to ThrowableAssert

### PR DESCRIPTION
**Pull request by librucha**

### CODE REVIEW

File: src/test/java/org/assertj/core/api/throwable/ThrowableAssert_hasMessageMatching_Test.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  @Test
+  public void should_fail_if_not_matching() throws Exception {
+    Throwable throwable = catchThrowable(raisingException("Given id='1' not exists"));
+
+    try {
+      assertThat(throwable).hasMessageMatching(REGEX);
+      failBecauseExceptionWasNotThrown(AssertionError.class);
+    } catch (AssertionError e) {
+      assertThat(e).hasMessage("\n"
```
Comment (PascalSchumacher) [inline src/test/java/org/assertj/core/api/throwable/ThrowableAssert_hasMessageMatching_Test.java]: Hi Libor,

can you replace `\n` with `String.format` and `%n` to be OS agnostic (see: https://github.com/joel-costigliola/assertj-core/issues/312)?

Thanks!

File: src/main/java/org/assertj/core/api/AbstractThrowableAssert.java [hunk 2]
```diff
@@ -126,6 +126,20 @@ public S hasMessageContaining(String description) {
   }
 
   /**
+   * Verifies that the message of the actual {@code Throwable} matches with the given regular expression.
+   *
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractThrowableAssert.java]: add a code example, thanks !

File: src/main/java/org/assertj/core/error/ShouldMatchingRegex.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  /**
+   * Creates a new <code>{@link ShouldMatchingRegex}</code>.
+   *
+   * @param actual the actual value in the failed assertion.
+   * @param regex the regular expression of value expected to be matched {@code actual}.
+   * @return the created {@code ErrorMessageFactory}.
+   */
+  public static ErrorMessageFactory shouldMatchingRegex(CharSequence actual, CharSequence regex) {
+    return new ShouldMatchingRegex("%nExpecting:%n  <%s>%nmatches regex:%n  <%s>%n", actual, regex);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldMatchingRegex.java]: The message should mention that we are checking an exception message : 
`"%nExpecting message:%n  <%s>%nto match regex:%n  <%s>%nbut did not.`
rename the class to `ShouldHaveMessageMatchingRegex`

**Comment by joel-costigliola** [comment]
thanks for the PR !

**Comment by joel-costigliola** [comment]
Integrated thanks

**Pull request closed without merging**